runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -246,15 +254,21 @@ public long append(Object object,
      * @param readStartTime start time of the range of reads.
      * @return log data at the address.
      */
+    @Deprecated
     protected ILogData read(final long address, long readStartTime) {
         try {
             if (System.currentTimeMillis() - readStartTime <
                     runtime.getParameters().getHoleFillTimeout().toMillis()) {
-                return runtime.getAddressSpaceView().read(address);
+                return runtime.getAddressSpaceView().read(address, readOptions);
             }
-            return runtime.getAddressSpaceView()
-                    .read(Collections.singleton(address), false, false)
-                    .get(address);
+
+            ReadOptions options = ReadOptions.builder()
```

Comment:
We can simplify the long build block here. 
You can pass a parameter  called `toBuilder` to ReadOption builder: `Builder(toBuilder=true)`, then here you can simply use: 
```java
ReadOptions options = readOptions.toBuilder()
        .waitForHole(false)
        .build()
```
This will make other fields same as `readOption` except for `waitForHole`

Reply:
Nice, didnt know about it.

Reply:
Is this done?

Reply:
+1 nice simplification.

Reply:
done.